sdk/core/azure-core-test/src/main/java/com/azure/core/test/annotation/PlaybackOnly.java

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.test.annotation;
+
+import com.azure.core.test.TestMode;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * Annotation given to some tests to indicate that it only runs when {@link TestMode} is {@link TestMode#PLAYBACK}.
+ * <p>
+ * When a test is annotated with this and {@link TestMode} is either {@link TestMode#LIVE} or {@link TestMode#RECORD}
+ * it will be skipped.
+ */
+@Retention(RUNTIME)
+@Target({METHOD})
+public @interface PlaybackOnly {
```

Comment:
In what scenarios would a test have to run only in playback mode? All playback tests should be able to run in LIVE and RECORD mode too.

Reply:
My thoughts for this was a test which is for a new service feature which is gated behind a flag, or worse requiring manual deployment, which wouldn't be possible during nightly live testing. Thinking on it more, it may be better to have a more nuance mechanism which falls back to a recording when the live test isn't able to run properly.

Reply:
We should have ARM templates to confirm canary config. See [Service Bus for example](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/servicebus/tests.yml#L11). Also, as you mentioned, we can provide an option to fall back to recording if the live test fails.

Reply:
This will be removed from this PR and likely re-worked as a concept where a test will fallback to a recording if live testing fails. The new concept could also support failing the build still when falling back but still test that code changes didn't break functionality.